packages/connect/src/base/entry.ts

```diff
@@ -236,9 +237,14 @@ export class EntryNodes extends EventEmitter {
       }
 
       // Ignore if entry nodes have more than one address
+      const firstUsableAddress = usableAddresses[0]
+
+      // Ignore if we're already connected to the address
+      if (this.usedRelays.some((usedRelay) => compareDirectConnectionInfo(usedRelay, firstUsableAddress))) continue
```

Comment:
`this.usedRelays` does not contain any direct addresses, so we cannot use it to check for already used IP:Port combinations